docs/design-doc.md

```diff
@@ -0,0 +1,296 @@
+      "definition_id": "32f54163-7166-48f1-93d8-ff217bdb0653",
+      "descriptor_map": [
+        {
+          "id": "id1",
+          "format": "jwt_vc",
+          "path": "$.verifiableCredential[0]"
+        }
+      ]
+    }
+  }
+}
+```
+
+The default behavior can be overridden in the config file by setting `handlers[x].endpoint.requestOptions.body` to a function. e.g:
+
+```javascript=
+async function transformRequestBody(dwm, data) {
+  // construct and return the request body you want sent to the downstream service
+}
+```
+
+# `Downstream Service -> DWN Relay` Response Payload
+The response sent from the Downstream Service back to the DWN Relay is evaluated against the `responseMapping` property for the respective handler that relayed the request.
+
+## Example
+
+### Config
+
+```json
+{
+  "handlers": [
+    {
+      "filter": { 
+        "method": "CollectionsWrite", 
+        "schema": "https://ssi-schemas.org/CredentialApplication",
+        "protocol": ""
+      },
+      "endpoint": {
+        "method": "POST",
+        "url": "https://somehost/some-endpoint",
+        "responseMapping": {
+          201: { "method": "CollectionsWrite", "schema": "https://ssi-schemas.org/CredentialApplication" }
+        }
+      }
+    }
+  ]
+}
+```
+
+### Response from Downstream Service
+```
+201: Created
+```
+**Response Body**
+```json
+{
+    "response": {
+        "id": "536eda73-b89f-4aef-8235-292c7c03ae85",
+        "spec_version": "https://identity.foundation/credential-manifest/spec/v1.0.0/",
+        "manifest_id": "WA-DL-CLASS-A",
+        "application_id": "9b1deb4d-3b7d-4bad-9bdd-2b0d7b3dcb6d",
+        "fulfillment": {
+            "descriptor_map": [
+                {
+                    "id": "id1",
+                    "format": "jwt_vc",
+                    "path": "$.verifiableCredential[0]"
+                }
+            ]
+        }
+    },
+    "credentials": [
+        {
+            "@context": [
+                "https://www.w3.org/2018/credentials/v1"
+            ],
+            "id": "a4a9e17b-b865-4cae-8085-ceb5584c18ea",
+            "type": [
+                "VerifiableCredential"
+            ],
+            "issuer": "did:example:123?linked-domains=3",
+            "issuanceDate": "2022-09-23T20:15:18Z",
+            "credentialSubject": {
+                "id": "did:user:123"
+            }
+        }
+    ]
+}
+```
+
+The DWN Relay Service will evaluate the response against the appropriate `responseMapping`. Based on the config example above, the Relay Service would: 
+- construct a DWeb Message using: 
+  - the response body sent by the Downstream Service in conjuction with properties 
+  - properties declared in `responseMapping` (e.g. `schema` and `method`)
+  - properties from the ingress DWeb Message that triggered the handler
+- Sign the message as per the [Signed Data](https://identity.foundation/decentralized-web-node/spec/#signed-data) section of the DWN spec
+- Store the message
```

Comment:
This can't happen if storage is turned off right?

Reply:
yup you're right good call out. I think i call this out somewhere in this design doc.